system-contracts/contracts/ContractDeployer.sol

```diff
@@ -385,6 +416,122 @@ contract ContractDeployer is IContractDeployer, SystemContractBase {
         }
     }
 
+    /// @notice Returns the address of the account that is delegated to execute transactions on behalf of the given
+    /// address.
+    /// @notice Returns the zero address if no delegation is set.
+    function getAccountDelegation(address _addr) public view override returns (address) {
+        bytes32 codeHash = ACCOUNT_CODE_STORAGE_SYSTEM_CONTRACT.getRawCodeHash(_addr);
+        if (codeHash[0] == 0x02 && codeHash[1] == 0x02) {
+            // The first two bytes of the code hash are 0x0202, which means that the account is delegated.
+            // The delegation address is stored in the last 20 bytes of the code hash.
+            return address(uint160(uint256(codeHash & DELEGATION_ADDRESS_MASK)));
+        } else {
+            // The account is not delegated.
+            return address(0);
+        }
+    }
+
+    /// @notice Method called by bootloader during processing of EIP7702 authorization lists.
+    /// @notice Each item is processed independently, so if any check fails for an item,
+    /// it is skipped and the next item is processed.
+    function processDelegations(AuthorizationListItem[] calldata authorizationList) external onlyCallFromBootloader {
+        uint256 listLength = authorizationList.length;
+        // The transaction is considered invalid if the length of authorization_list is zero.
+        if (listLength == 0) {
+            revert EmptyAuthorizationList();
+        }
+        for (uint256 i = 0; i < listLength; ++i) {
+            // Per EIP7702 rules, if any check for the tuple item fails,
+            // we must move on to the next item in the list.
+            AuthorizationListItem calldata item = authorizationList[i];
+
+            // Verify the chain ID is 0 or the ID of the current chain.
+            if (item.chainId != 0 && item.chainId != block.chainid) {
+                continue;
+            }
+
+            // Verify the nonce is less than 2**64.
+            if (item.nonce >= 2 ** 64) {
```

Comment:
Nit: Off-by-1 error?
As per EIP: we have to verify the nonce is *less* than 2**64 - 1.
I assume that means strictly less.